### PR TITLE
Louis/alpine matching

### PR DESCRIPTION
**Pull request by ldelossa**
This PR introduces alpine vulnerability matching. 
A few errors were encountered along with a sql query regression. This PR overflows a bit to account for these fixes.

* Introduces a lookup index back on the vuln table. This improves the matching speed by several orders of magnitude.
* Doubles the buffer size on the bufio.Scanner in the alpine scanner due to running into "token to large" issue while parsing
* Add PrettyName support in os-release and distribution struct. Plumb this and CPE fields thru libvuln and into database
* Update tests to handle above
* And finally... implement alpine matcher interface

### CODE REVIEW

File: internal/vulnstore/postgres/putvulnerabilities.go [hunk 1]
```diff
@@ -83,10 +34,12 @@ const (
 				  dist_version,
 				  dist_version_code_name,
 				  dist_version_id,
+			      dist_arch,
+			      dist_cpe,
+			      dist_pretty_name,
```
Comment (hdonnay) [inline internal/vulnstore/postgres/putvulnerabilities.go]: formatting

File: internal/vulnstore/postgres/putvulnerabilities.go [hunk 2]
```diff
@@ -121,10 +76,12 @@ const (
 				 dist_version,
 				 dist_version_code_name,
 				 dist_version_id,
+			     dist_arch,
+			     dist_cpe,
+			     dist_pretty_name,
```
Comment (hdonnay) [inline internal/vulnstore/postgres/putvulnerabilities.go]: formatting

File: internal/vulnstore/postgres/querybuilder_test.go [hunk 3]
```diff
@@ -85,10 +85,14 @@ func Test_GetBuilder_Determinist_Arguments(t *testing.T) {
 	for _, table := range tt {
 		t.Run(table.name, func(t *testing.T) {
 			q, _, err := getBuilder(table.matchers)
-			assert.NoError(t, err)
+			if err != nil {
+				t.Fatalf("failed to build query: %v", err)
+			}
 
 			// assert.Equal(t, table.expected, q)
```
Comment (hdonnay) [inline internal/vulnstore/postgres/querybuilder_test.go]: delete commented code

File: alpine/matcher.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package alpine
+
+import (
+	version "github.com/knqyf263/go-deb-version"
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply alpine/matcher.go]: Believe it does, this code works along with the fact ClairV2 uses the same hand crafted dpkg parser for alpine also. 

I was looking for documentation stating this for a bit and couldn't find anything tho. I did some manual testing and `3.14.0-r0` < `3.14.0.r1` so the non semver portions are handled correctly.

File: alpine/packagescanner.go [hunk 5]
```diff
@@ -85,6 +86,8 @@ func (*Scanner) ScanContext(ctx context.Context, layer *claircore.Layer) ([]*cla
 	// can't because the database "keys" are case sensitive, unlike MIME
 	// headers. So, roll our own entry and header splitting.
 	s := bufio.NewScanner(bytes.NewReader(b))
+	buf := make([]byte, 0, maxTokenSize)
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply alpine/packagescanner.go]: was thinking that too - and then al scanners would use the same pool ?

[the remark this answers is not in the document]
Comment (ldelossa) [reply alpine/packagescanner.go]: @hdonnay so a problem with passing in the pool is that these package scanners are interface implementations. We'd need to inject the pool somewhere. The Scan interface method comes to mind but it maybe a larger PR.

[the remark this answers is not in the document]
Comment (hdonnay) [reply alpine/packagescanner.go]: it can just be a package-level pool

**Pull request merged**